aiosmtpd/smtp.py

```diff
@@ -381,42 +404,64 @@ def __init__(
                 log.warning("proxy_protocol_timeout < 3.0")
         self._proxy_timeout = proxy_protocol_timeout
 
-        self._authenticator: Optional[AuthenticatorType]
-        self._auth_callback: Optional[AuthCallbackType]
         if authenticator is not None:
             self._authenticator = authenticator
             self._auth_callback = None
         else:
             self._auth_callback = auth_callback or login_always_fail
             self._authenticator = None
-
         self._auth_required = auth_required
+        self._auth_excludes = auth_exclude_mechanism
 
-        # Get hooks & methods to significantly speedup getattr's
-        self._auth_methods: Dict[str, _AuthMechAttr] = {
-            getattr(
-                mfunc, "__auth_mechanism_name__",
-                mname.replace("auth_", "").replace("__", "-")
-            ): _AuthMechAttr(mfunc, obj is self)
-            for obj in (self, handler)
-            for mname, mfunc in inspect.getmembers(obj)
-            if mname.startswith("auth_")
-        }
```

Comment:
Moving method scanning outside of __init__, especially for a scanning as complex as this, speeds up SMTP instantiation.